### PR TITLE
Added ./sync and ./types to lint packages

### PR DESCRIPTION
**Pull request by mharkus**
1. I need some help in adding comments with some of the exported variables identified by {help}.
2. go fmt removes the tab spaces in variable declarations so right hand side assignments will not be vertically aligned anymore. Let me know if this is ok, otherwise, I will format them back to the original alignment.

for #2702

### CODE REVIEW

File: types/block.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ type (
 		Transactions []Transaction   `json:"transactions"`
 	}
 
-	// A BlockHeader, when encoded, is an 80-byte constant size field
+	// BlockHeader when encoded, is an 80-byte constant size field
```
Comment (lukechampine) [inline types/block.go]: Interesting. `golint` doesn't care about the `A`, but it does care about the `,`. I would rephrase this as:
```
A BlockHeader contains the data that, when hashed, produces the Block's ID.
```

File: types/block.go [hunk 2]
```diff
@@ -42,9 +42,12 @@ type (
 		MerkleRoot crypto.Hash `json:"merkleroot"`
 	}
 
+	// BlockHeight refers to the height of the block
```
Comment (lukechampine) [inline types/block.go]: more specifically,
```
BlockHeight is the number of blocks between a given block and the Genesis block.
The BlockHeight of the Genesis block is 0.
```

File: types/block.go [hunk 3]
```diff
@@ -42,9 +42,12 @@ type (
...
 	BlockHeight uint64
-	BlockID     crypto.Hash
-	BlockNonce  [8]byte
+	// BlockID is an identifier used to identify the block
+	BlockID crypto.Hash
+	// BlockNonce {help}
+	BlockNonce [8]byte
```
Comment (lukechampine) [inline types/block.go]:
```
The BlockNonce is a "scratch space" that miners can freely alter to produce
a BlockID that satisfies a given Target.
```

File: types/block.go [hunk 4]
```diff
@@ -42,9 +42,12 @@ type (
 		MerkleRoot crypto.Hash `json:"merkleroot"`
 	}
 
+	// BlockHeight refers to the height of the block
 	BlockHeight uint64
-	BlockID     crypto.Hash
-	BlockNonce  [8]byte
+	// BlockID is an identifier used to identify the block
+	BlockID crypto.Hash
```
Comment (lukechampine) [inline types/block.go]:
```
A BlockID is the hash of a BlockHeader. A BlockID uniquely
identifies a Block, and indicates the amount of work performed
to mine that Block. The more leading zeros in the BlockID, the
more work was performed.
```

File: types/constants.go [hunk 5]
```diff
@@ -13,42 +13,70 @@ import (
 )
 
 var (
-	BlockFrequency         BlockHeight
-	BlockSizeLimit         = uint64(2e6)
+	// BlockFrequency {help}
+	BlockFrequency BlockHeight
```
Comment (lukechampine) [inline types/constants.go]:
```
BlockFrequency is the desired number of seconds that
should elapse, on average, between successive Blocks.
```

File: types/constants.go [hunk 6]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// BlockSizeLimit {help}
+	BlockSizeLimit = uint64(2e6)
```
Comment (lukechampine) [inline types/constants.go]:
```
BlockSizeLimit is the maximum size of a binary-encoded Block
that is permitted by the consensus rules.
```

File: types/constants.go [hunk 7]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// ExtremeFutureThreshold {help}
 	ExtremeFutureThreshold Timestamp
```
Comment (lukechampine) [inline types/constants.go]:
```
ExtremeFutureThreshold is a temporal limit beyond which Blocks are
discarded by the consensus rules. When incoming Blocks are processed, their
Timestamp is allowed to exceed the processor's current time by a small amount.
But if the Timestamp is further into the future than ExtremeFutureThreshold,
the Block is immediately discarded.
```

File: types/constants.go [hunk 8]
```diff
@@ -13,42 +13,70 @@ import (
...
-	FutureThreshold        Timestamp
-	GenesisBlock           Block
+	// FutureThreshold {help}
+	FutureThreshold Timestamp
```
Comment (lukechampine) [inline types/constants.go]:
```
ExtremeFutureThreshold is a temporal limit beyond which Blocks are
discarded by the consensus rules. When incoming Blocks are processed, their
Timestamp is allowed to exceed the processor's current time by no more than
FutureThreshold. If the excess duration is larger than FutureThreshold, but
smaller than ExtremeFutureThreshold, the Block may be held in memory until
the Block's Timestamp exceeds the current time by less than FutureThreshold.
```

File: types/constants.go [hunk 9]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// GenesisBlock is the first block of the block chain
+	GenesisBlock Block
 
-	// The GenesisID is used in many places. Calculating it once saves lots of
+	// GenesisID is used in many places. Calculating it once saves lots of
```
Comment (lukechampine) [inline types/constants.go]: articles like `The` should be allowed by golint

File: types/constants.go [hunk 10]
```diff
@@ -13,42 +13,70 @@ import (
...
 	// redundant computation.
 	GenesisID BlockID
 
+	// GenesisSiafundAllocation {help}
 	GenesisSiafundAllocation []SiafundOutput
```
Comment (lukechampine) [inline types/constants.go]:
```
GenesisSiafundAllocation is the set of SiafundOutputs created in the Genesis block.
```

File: types/constants.go [hunk 11]
```diff
@@ -13,42 +13,70 @@ import (
...
-	InitialCoinbase          = uint64(300e3)
-	MaturityDelay            BlockHeight
-	MaxAdjustmentDown        *big.Rat
-	MaxAdjustmentUp          *big.Rat
-	MedianTimestampWindow    = uint64(11)
-	MinimumCoinbase          uint64
+	// GenesisTimestamp is the timestamp when genesis block was mined
+	GenesisTimestamp Timestamp
+	// InitialCoinbase {help}
+	InitialCoinbase = uint64(300e3)
```
Comment (lukechampine) [inline types/constants.go]:
```
InitialCoinbase is the coinbase reward of the Genesis block.
```

File: types/constants.go [hunk 12]
```diff
@@ -13,42 +13,70 @@ import (
...
+	RootDepth = Target{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
+	// RootTarget {help}
+	RootTarget Target
+	// SiacoinPrecision {help}
 	SiacoinPrecision = NewCurrency(new(big.Int).Exp(big.NewInt(10), big.NewInt(24), nil))
-	SiafundCount     = NewCurrency64(10000)
-	SiafundPortion   = big.NewRat(39, 1000)
-	TargetWindow     BlockHeight
+	// SiafundCount {help}
+	SiafundCount = NewCurrency64(10000)
```
Comment (lukechampine) [inline types/constants.go]:
```
SiafundCount is the total number of Siafunds in existence.
```

File: types/constants.go [hunk 13]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// SiafundPortion {help}
+	SiafundPortion = big.NewRat(39, 1000)
```
Comment (lukechampine) [inline types/constants.go]:
```
SiafundPortion is the percentage of siacoins that is taxed from FileContracts.
```

File: types/filecontracts.go [hunk 14]
```diff
@@ -90,6 +92,7 @@ type (
 		HashSet  []crypto.Hash            `json:"hashset"`
 	}
 
+	// ProofStatus {help}
 	ProofStatus bool
```
Comment (lukechampine) [inline types/filecontracts.go]:
```
ProofStatus indicates whether a StorageProof was valid (true) or missed (false).
```

File: types/filecontracts.go [hunk 15]
```diff
@@ -9,8 +9,10 @@ import (
 )
 
 var (
+	// ProofMissed {help}
 	ProofMissed ProofStatus = false
```
Comment (lukechampine) [inline types/filecontracts.go]:
```
ProofMissed indicates that a StorageProof was missed, which means that
no valid proof was submitted within the proof window.
```

File: types/filecontracts.go [hunk 16]
```diff
@@ -9,8 +9,10 @@ import (
...
-	ProofValid  ProofStatus = true
+	// ProofValid {help}
+	ProofValid ProofStatus = true
```
Comment (lukechampine) [inline types/filecontracts.go]:
```
ProofValid indicates that a valid StorageProof was submitted within the proof window.
```

File: types/target.go [hunk 17]
```diff
@@ -43,16 +45,16 @@ func (x Target) Cmp(y Target) int {
 }
 
 // Difficulty returns the difficulty associated with a given target.
-func (t Target) Difficulty() Currency {
-	if t == (Target{}) {
+func (x Target) Difficulty() Currency {
+	if x == (Target{}) {
```
Comment (lukechampine) [inline types/target.go]: these should use `t` throughout instead of `x`

File: types/currency.go [hunk 18]
```diff
@@ -158,8 +158,8 @@ func (x Currency) RoundDown(y Currency) (c Currency) {
 }
 
 // IsZero returns true if the value is 0, false otherwise.
-func (c Currency) IsZero() bool {
-	return c.i.Sign() <= 0
+func (x Currency) IsZero() bool {
```
Comment (lukechampine) [inline types/currency.go]: these should use `c` throughout instead of `x`

File: types/timestamp.go [hunk 19]
```diff
@@ -8,7 +8,10 @@ import (
 )
 
 type (
-	Timestamp      uint64
+	// Timestamp is the digital record of the time of occurrence of a particular
+	// event
+	Timestamp uint64
```
Comment (lukechampine) [inline types/timestamp.go]:
```
Timestamp is a Unix timestamp, i.e. the number of the seconds since Jan 1 1970.
```

File: types/transactions.go [hunk 20]
```diff
@@ -14,6 +14,7 @@ import (
 )
 
 const (
+	// SpecifierLen {help}
 	SpecifierLen = 16
```
Comment (lukechampine) [inline types/transactions.go]:
```
SpecifierLen is the length in bytes of a Specifier.
```

File: types/transactions.go [hunk 21]
```diff
@@ -59,11 +60,20 @@ type (
...
+	// SiafundOutputID refers to the identification used to identify a siafund
+	// output
 	SiafundOutputID crypto.Hash
-	FileContractID  crypto.Hash
-	OutputID        crypto.Hash
+	// FileContractID refers to the identification used to identify a file
+	// contract
+	FileContractID crypto.Hash
+	// OutputID refers to the identification used to identify an output
+	OutputID crypto.Hash
```
Comment (lukechampine) [inline types/transactions.go]: For these, instead of `ThingID refers to the identification used to identify a Thing`, I would say `A ThingID uniquely identifies a Thing`.

File: types/constants.go [hunk 22]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// MaturityDelay {help}
+	MaturityDelay BlockHeight
+	// MaxAdjustmentDown {help}
+	MaxAdjustmentDown *big.Rat
+	// MaxAdjustmentUp {help}
+	MaxAdjustmentUp *big.Rat
+	// MedianTimestampWindow {help}
+	MedianTimestampWindow = uint64(11)
+	// MinimumCoinbase {help}
+	MinimumCoinbase uint64
```
Comment (lukechampine) [inline types/constants.go]:
```
MinimumCoinbase is the minimum coinbase reward for a block.
The coinbase decreases in each block after the Genesis block,
but it will not decrease past MinimumCoinbase.
```

File: types/target.go (hunk 17 above)
Comment (mharkus) [reply types/target.go]: So for functions like this:

func (x Target) SubtractDifficulties(y Target) (t Target)

this needs to be changed to:

func (t Target) SubtractDifficulties(y Target) (x Target)

but does this also mean that callers of this function have to change as well now that the return variable changes? or that doesn't really matter?

It's complaining about duplicate argument if I leave them both as t.

File: types/constants.go (hunk 9 above)
Comment (mharkus) [reply types/constants.go]: golint follows a certain format for comments and so it fails with this message when linting:

types/constants.go:38:2: comment on exported var GenesisID should be of the form "GenesisID ..."

Comment (mharkus) [reply types/constants.go]: apparently, constants are not allowed to have leading articles: https://github.com/golang/lint/issues/11. That's why it worked on some comments, for example: in block.go. BlockHeader was defined as type whereas in constants.go, GenesisBlock was defined as var.

**Comment by mharkus** [comment]
@lukechampine I tried to do this:

        // Other consensus variables
	MaturityDelay         BlockHeight
	MaxAdjustmentDown     *big.Rat
	MaxAdjustmentUp       *big.Rat
	MedianTimestampWindow = uint64(11)

but still getting this when linting:

types/constants.go:49:2: comment on exported var MaturityDelay should be of the form "MaturityDelay ..."
types/constants.go:51:2: exported var MaxAdjustmentDown should have comment or be unexported

Should we just add TODO to those vars that cannot be commented on right now?

Something like:

// OakDecayDenom TODO: give description
OakDecayDenom int64

File: types/constants.go (hunk 9 above)
Comment (lukechampine) [reply types/constants.go]: huh. That's strange, but good to know.

File: types/target.go (hunk 17 above)
Comment (lukechampine) [reply types/target.go]: golint just wants the name of the receiving variable to be consistent across methods. So it should be `func (t Target) Foo` for every method. Callers of the function shouldn't be affected.

**Comment by lukechampine** [comment]
drat. Okay, I'll go through and come up with descriptions for the rest.

File: Makefile [hunk 23]
```diff
@@ -52,9 +52,10 @@ vet: release-std
 
 # will always run on some packages for a while.
 lintpkgs = ./build ./cmd/siac ./cmd/siad ./compatibility ./crypto ./encoding ./modules ./modules/consensus ./modules/explorer \
-           ./modules/gateway ./modules/host ./modules/miner ./modules/host/contractmanager ./modules/renter ./modules/renter/contractor ./modules/renter/hostdb \
-           ./modules/renter/hostdb/hosttree ./modules/renter/proto ./modules/wallet ./modules/transactionpool ./node ./node/api ./node/api/server ./persist \
-           ./siatest ./siatest/consensus ./siatest/renter
+           ./modules/gateway ./modules/host ./modules/miner ./modules/host/contractmanager ./modules/renter 				          \
+		       ./modules/renter/contractor ./modules/renter/hostdb ./modules/renter/hostdb/hosttree ./modules/renter/proto        \
+		       ./modules/wallet ./modules/transactionpool ./node ./node/api ./node/api/server ./persist ./siatest                 \
```
Comment (DavidVorick) [inline Makefile]: tabs vs. spaces issue here

File: types/constants.go [hunk 24]
```diff
@@ -13,42 +13,70 @@ import (
...
-	MaxAdjustmentDown        *big.Rat
-	MaxAdjustmentUp          *big.Rat
-	MedianTimestampWindow    = uint64(11)
-	MinimumCoinbase          uint64
+	// GenesisTimestamp is the timestamp when genesis block was mined
+	GenesisTimestamp Timestamp
+	// InitialCoinbase {help}
+	InitialCoinbase = uint64(300e3)
+	// MaturityDelay {help}
+	MaturityDelay BlockHeight
```
Comment (DavidVorick) [inline types/constants.go]: MaturityDelay specifies the number of blocks that a maturity-required output is required to be on hold before it can be spent on the blockchain. Outputs are maturity-required if they are highly likely to be altered or invalidated in the event of a small reorg. One example is the block reward, as a small reorg may invalidate the block reward. Another example is a siafund payout, as a tiny reorg may change the value of the payout, and thus invalidate any transactions spending the payout. File contract payouts also are subject to a maturity delay.

File: types/constants.go [hunk 25]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// MaxAdjustmentDown {help}
```
Comment (DavidVorick) [inline types/constants.go]: This should probably be changed to `MaxTargetAdjustmentDown`, and the other one `MaxTargetAdjustmentUp`.

These values restrict how much the block difficulty is allowed to change in a single step, which is important to limit the effect of difficulty raising and lowering attacks.

File: types/constants.go [hunk 26]
```diff
@@ -13,42 +13,70 @@ import (
...
+	MaxAdjustmentDown *big.Rat
+	// MaxAdjustmentUp {help}
+	MaxAdjustmentUp *big.Rat
+	// MedianTimestampWindow {help}
+	MedianTimestampWindow = uint64(11)
```
Comment (DavidVorick) [inline types/constants.go]: `MedianTimestampWindow` tells us how many blocks to look back when calculating the median timestamp over the previous `n` blocks. The timestamp of a block is not allowed to be less than or equal to the median timestamp of the previous `n` blocks, where for Sia this number is typically 11.

File: types/constants.go [hunk 27]
```diff
@@ -13,42 +13,70 @@ import (
...
 
 	// Oak hardfork constants. Oak is the name of the difficulty algorithm for
 	// Sia following a hardfork at block 135e3.
-	OakDecayDenom           int64
-	OakDecayNum             int64
-	OakHardforkBlock        BlockHeight
-	OakHardforkFixBlock     BlockHeight
+
+	// OakDecayDenom {help}
+	OakDecayDenom int64
```
Comment (DavidVorick) [inline types/constants.go]: The denominator for how much the total timestamp is decayed each step.

File: types/constants.go [hunk 28]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// OakDecayNum {help}
+	OakDecayNum int64
```
Comment (DavidVorick) [inline types/constants.go]: The numerator for how much the total timestamp is decayed each step.

File: types/constants.go [hunk 29]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// OakHardforkBlock {help}
+	OakHardforkBlock BlockHeight
```
Comment (DavidVorick) [inline types/constants.go]: The height at which the hardfork to switch to the oak difficulty adjustment algorithm is triggered.

File: types/constants.go [hunk 30]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// OakHardforkFixBlock {help}
+	OakHardforkFixBlock BlockHeight
```
Comment (DavidVorick) [inline types/constants.go]: The height at which the hardfork to switch from the broken oak difficulty adjustment algorithm to the fixed oak difficulty adjustment algorithm is triggered.

File: types/constants.go [hunk 31]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// OakHardforkTxnSizeLimit {help}
 	OakHardforkTxnSizeLimit = uint64(64e3) // 64 KB
```
Comment (DavidVorick) [inline types/constants.go]: The maximum size allowed for a transaction, a change which I believe was implemented simultaneously with the oak hardfork.

File: types/constants.go [hunk 32]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// OakMaxRise {help}
+	OakMaxRise *big.Rat
 
-	RootDepth        = Target{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
-	RootTarget       Target
+	// RootDepth {help}
+	RootDepth = Target{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
+	// RootTarget {help}
+	RootTarget Target
+	// SiacoinPrecision {help}
```
Comment (DavidVorick) [inline types/constants.go]: The number of base units in a siacoin. The Sia network has a very large number of base units. We call 10^24 of these a siacoin.

The base unit for Bitcoin is called a satoshi. We call 10^8 satoshis a bitcoin, even though the code itself only ever works with satoshis.

File: types/constants.go [hunk 33]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// OakMaxDrop {help}
+	OakMaxDrop *big.Rat
+	// OakMaxRise {help}
+	OakMaxRise *big.Rat
 
-	RootDepth        = Target{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
-	RootTarget       Target
+	// RootDepth {help}
+	RootDepth = Target{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
+	// RootTarget {help}
```
Comment (DavidVorick) [inline types/constants.go]: The root target is the target for the genesis block - basically how much work needs to be done in order to mine the first block. The difficulty adjustment algorithm takes over from there.

File: types/constants.go [hunk 34]
```diff
@@ -13,42 +13,70 @@ import (
...
+	OakMaxBlockShift int64
+	// OakMaxDrop {help}
+	OakMaxDrop *big.Rat
+	// OakMaxRise {help}
+	OakMaxRise *big.Rat
 
-	RootDepth        = Target{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
-	RootTarget       Target
+	// RootDepth {help}
+	RootDepth = Target{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
```
Comment (DavidVorick) [inline types/constants.go]: The depth is the cumulative target of all blocks. The root depth is essentially the maximum possible target, there have been no blocks yet, so there is no cumulated difficulty yet.

File: types/constants.go [hunk 35]
```diff
@@ -13,42 +13,70 @@ import (
...
+	// OakHardforkTxnSizeLimit {help}
 	OakHardforkTxnSizeLimit = uint64(64e3) // 64 KB
-	OakMaxBlockShift        int64
-	OakMaxDrop              *big.Rat
-	OakMaxRise              *big.Rat
+	// OakMaxBlockShift {help}
+	OakMaxBlockShift int64
+	// OakMaxDrop {help}
+	OakMaxDrop *big.Rat
+	// OakMaxRise {help}
```
Comment (DavidVorick) [inline types/constants.go]: The shift is the maximum number of seconds that the oak algorithm will shift the difficulty.

The drop is the maximum amount that the difficulty will drop each block, and the rise is the maximum amount that the difficulty will rise each block.

**Comment by DavidVorick** [comment]
@mharkus I think we've filled them all out at this point, let us know if we missed anything

**Comment by DavidVorick** [comment]
*or if anything needs additional clarification.

Sorry for closing this PR, hit the wrong button by mistake.

**Comment by mharkus** [comment]
Ok thanks! I will put all the comments above and update the PR.

**Comment by tbenz9** [comment]
@mharkus bump.  :)

**Comment by mharkus** [comment]
Hi @DavidVorick The TargetWindow BlockHeight is still missing a comment. Also, I will create a separate PR for the renaming of MaxAdjustmentDown and MaxAdjustmentUp.

File: types/constants.go [hunk 36]
```diff
@@ -13,42 +13,113 @@ import (
...
 	SiacoinPrecision = NewCurrency(new(big.Int).Exp(big.NewInt(10), big.NewInt(24), nil))
-	SiafundCount     = NewCurrency64(10000)
-	SiafundPortion   = big.NewRat(39, 1000)
-	TargetWindow     BlockHeight
+	// SiafundCount is the total number of Siafunds in existence.
+	SiafundCount = NewCurrency64(10000)
+	// SiafundPortion is the percentage of siacoins that is taxed from FileContracts.
+	SiafundPortion = big.NewRat(39, 1000)
+	// TargetWindow {help}
+	TargetWindow BlockHeight
```
Comment (DavidVorick) [inline types/constants.go]: TargetWindow is the number of blocks to look backwards when determining how much time has passed vs. how many blocks have been created. It's only used in the old, broken difficulty adjustment algorithm.

File: types/block.go [hunk 37]
```diff
@@ -32,19 +32,25 @@ type (
...
+	// A BlockHeader contains the data that, when hashed, produces the Block's ID.
 	BlockHeader struct {
 		ParentID   BlockID     `json:"parentid"`
 		Nonce      BlockNonce  `json:"nonce"`
 		Timestamp  Timestamp   `json:"timestamp"`
 		MerkleRoot crypto.Hash `json:"merkleroot"`
 	}
 
+	// BlockHeight is the number of blocks between a given block and the Genesis block.
+	// The BlockHeight of the Genesis block is 0.
```
Comment (DavidVorick) [inline types/block.go]: BlockHeight is the number of blocks that exist after the genesis block.

**Comment by DavidVorick** [comment]
Added a description for TargetWindow. Looks like BlockHeight already had a description, but I clarified it.

Thanks for taking the time to clean up this last, monster package.
